ssl/statem/statem_srvr.c

```diff
@@ -3616,6 +3661,18 @@ int tls_construct_server_certificate(SSL *s, WPACKET *pkt)
     return 1;
 }
 
+#if !defined(OPENSSL_NO_COMP)
+int tls_construct_server_compressed_certificate(SSL *s, WPACKET *pkt)
+{
+    if (s->s3.tmp.cert == NULL) {
```

Comment:
No, the state machine has to be able to create a compressed certificate message or a regular certificate message depending on the ClientHello. Just because the server may do a regular certificate message in one connection does not mean that it will be done in all connections. Both functions need to exist simultaneously.